### PR TITLE
[NPU] Support dp-attention for MiniMax2.5

### PR DESCRIPTION
**Pull request by shadowxz109**
## Motivation
Support dp-attention for MiniMax2.5

## Modifications
Support dp-attention for MiniMax2.5
Resolve the issue that fused_topk_native does not support num_token_non_padded is not None.

## Accuracy Tests
gsm8k test:
Accuracy: 0.955
Invalid: 0.000
Latency: 34.479 s
Output throughput: 566.869 token/s

## Benchmarking and Profiling
npu cmd:
```
export SGLANG_NPU_FUSED_MOE_MODE=2
export SGLANG_DEEPEP_NUM_MAX_DISPATCH_TOKENS_PER_RANK=262144
export PYTORCH_NPU_ALLOC_CONF=expandable_segments:True

python -m sglang.launch_server --model-path $MODEL_PATH \
    --host 127.0.0.1 --port 32000 --trust-remote-code --nnodes 1 --node-rank 0  \
    --attention-backend ascend --device npu \
    --max-running-requests 256 \
    --tp 16 --mem-fraction-static 0.8 --cuda-graph-bs 1 8 16 32 48 52 56 60 64 72 \
    --dp-size 2 \
    --enable-dp-attention \
    --ep-size 16 --moe-a2a-backend ascend_fuseep --deepep-mode auto \
    --quantization modelslim
```

before: 
```
============ Serving Benchmark Result ============
Backend:                                 sglang    
Traffic request rate:                    inf       
Max request concurrency:                 96        
Successful requests:                     384       
Benchmark duration (s):                  270.99    
Total input tokens:                      1572864   
Total input text tokens:                 1572864   
Total generated tokens:                  384000    
Total generated tokens (retokenized):    383992    
Request throughput (req/s):              1.42      
Input token throughput (tok/s):          5804.23   
Output token throughput (tok/s):         1417.05   
Peak output token throughput (tok/s):    2208.00   
Peak concurrent requests:                144       
Total token throughput (tok/s):          7221.27   
Concurrency:                             91.10     
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   64288.35  
Median E2E Latency (ms):                 58113.10  
P90 E2E Latency (ms):                    88431.85  
P99 E2E Latency (ms):                    93468.46  
---------------Time to First Token----------------
Mean TTFT (ms):                          3863.06   
Median TTFT (ms):                        2750.93   
P99 TTFT (ms):                           13411.05  
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          60.49     
Median TPOT (ms):                        55.38     
P99 TPOT (ms):                           91.92     
---------------Inter-Token Latency----------------
Mean ITL (ms):                           60.49     
Median ITL (ms):                         44.65     
P95 ITL (ms):                            124.52    
P99 ITL (ms):                            148.11    
Max ITL (ms):                            12564.56
=========================================
```
after:
```
============ Serving Benchmark Result ============
Backend:                                 sglang    
Traffic request rate:                    inf       
Max request concurrency:                 96        
Successful requests:                     384       
Benchmark duration (s):                  236.92    
Total input tokens:                      1572864   
Total input text tokens:                 1572864   
Total generated tokens:                  384000    
Total generated tokens (retokenized):    383088    
Request throughput (req/s):              1.62      
Input token throughput (tok/s):          6638.70   
Output token throughput (tok/s):         1620.78   
Peak output token throughput (tok/s):    2400.00   
Peak concurrent requests:                192       
Total token throughput (tok/s):          8259.47   
Concurrency:                             95.96     
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   59204.95  
Median E2E Latency (ms):                 59202.12  
P90 E2E Latency (ms):                    59744.48  
P99 E2E Latency (ms):                    59762.14  
---------------Time to First Token----------------
Mean TTFT (ms):                          9422.14   
Median TTFT (ms):                        9406.10   
P99 TTFT (ms):                           17773.39  
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          49.83     
Median TPOT (ms):                        49.82     
P99 TPOT (ms):                           58.13     
---------------Inter-Token Latency----------------
Mean ITL (ms):                           49.83     
Median ITL (ms):                         41.37     
P95 ITL (ms):                            47.15     
P99 ITL (ms):                            49.11     
Max ITL (ms):                            17044.62 
============ Serving Benchmark Result ============
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/moe/topk.py [hunk 1]
```diff
@@ -39,33 +39,23 @@ def fused_topk_npu(
...
-            expert_location_dispatch_info=expert_location_dispatch_info,
+            "eps": float(1e-20),
+        }
+        if use_grouped_topk:
+            kwargs["k_group"] = topk_config.topk_group
+            kwargs["group_count"] = topk_config.num_expert_group
+            kwargs["group_select_mode"] = 1
+        topk_weights, topk_ids, _ = torch.ops.npu.npu_moe_gating_top_k(
+            router_logits.to(torch.float32), **kwargs
         )
```
Comment (McZyWu) [inline python/sglang/srt/hardware_backend/npu/moe/topk.py:91]: select_experts method is necessary for the case that custom_routing_function is not None.

Comment (shadowxz109) [reply python/sglang/srt/hardware_backend/npu/moe/topk.py:91]: ok, I have fixed it.

**Review by iforgetmyname — APPROVED** [review]
